.github/workflows/build.yml

```diff
@@ -456,6 +456,112 @@ jobs:
           ../scripts/xcodebuild.sh ${{ steps.configure.outputs.project-directory }}/TemplateExample.xcworkspace build
         working-directory: template-example
     timeout-minutes: 60
+  visionos:
+    name: "visionOS"
+    runs-on: macos-14
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          filter: blob:none
+          fetch-depth: 0
+      - name: Set up toolchain
+        uses: ./.github/actions/setup-toolchain
+        with:
+          platform: visionos
+          project-root: example
+          cache-key-prefix: example
+          xcode-developer-dir: /Applications/Xcode_15.2.app
+      - name: Set up react-native@nightly
+        if: ${{ github.event_name == 'schedule' }}
+        uses: ./.github/actions/setup-react-native
+        with:
+          version: nightly
+      - name: Install npm dependencies
+        uses: ./.github/actions/yarn
+        with:
+          immutable: ${{ github.event_name != 'schedule' }}
+      - name: Bundle JavaScript
+        run: |
+          yarn build:visionos || yarn build:visionos
+        working-directory: example
+      - name: Determine whether the visionOS app needs to be built
+        id: affected
+        uses: ./.github/actions/affected
+      - name: Install Pods
+        if: ${{ steps.affected.outputs.visionos != '' }}
+        uses: ./.github/actions/cocoapods
+        with:
+          project-directory: visionos
+          working-directory: example
+      - name: Build
+        if: ${{ steps.affected.outputs.visionos != '' }}
+        run: |
+          ../scripts/xcodebuild.sh visionos/Example.xcworkspace build-for-testing
+        working-directory: example
+      - name: Test `react-native config`
+        if: ${{ steps.affected.outputs.visionos != '' }}
+        run: |
+          node --test test/config.test.mjs
+        working-directory: example
+      # Disable as the VisionOS Simator seems to crash in CI
+      # - name: Test
+      #   if: ${{ steps.affected.outputs.visionos != '' }}
+      #   run: |
+      #     ../scripts/xcodebuild.sh visionos/Example.xcworkspace test-without-building
+      #   working-directory: example
```

Comment:
The iOS tests just make sure we correctly decode JSON files. We can skip this entirely since we'll be using the same code.

```suggestion
```